src/view/map_view.py

```diff
@@ -0,0 +1,105 @@
+import json
+import os
+from typing import Any
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib.animation import FuncAnimation
+from PIL import Image
+
+
+class ViewMap:
+    def __init__(self, map_name: str, csv_file_path: str):
+        # Get the directory of the current script
+        self.base_dir = os.path.dirname(os.path.abspath(__file__))
+
+        self.map_name = map_name
+        self.csv_file_path = os.path.join(self.base_dir, csv_file_path)
+        self.image_file_path = os.path.join(self.base_dir, "..", "assets", "maps", map_name, "display_icon.png")
+        self.json_file_path = os.path.join(self.base_dir, "..", "data", "maps", f"{map_name}.json")
+
+        self.selected_map = self.load_map_data()
```

Comment:
GameMap class already has a `from_json()` method that reads the json file and inits a class
```suggestion
        self.selected_map = GameMap.from_json(self,json_file_path)
```

maybe we can add the if not path and error handling logic from load_map_data() method to from_json() one

Reply:
Yes, I totally agree. I added this try/catch block to improve robustness. I will update the GameMap class with this try/catch as well.

Reply:
Solved